src/vm/vm_memory/memory.rs

```diff
@@ -113,6 +113,20 @@ impl Memory {
         }
         Ok(())
     }
+
+    pub fn get_range(
+        &self,
+        addr: &MaybeRelocatable,
+        size: usize,
+    ) -> Result<Vec<Option<&MaybeRelocatable>>, MemoryError> {
+        let mut values = Vec::new();
+
+        for i in 0..size {
+            values.push(self.get(&addr.add_usize_mod(i, None))?);
+        }
+
+        Ok(values)
+    }
```

Comment:
I'd much rather have an iterator return elements one by one. But even if we keep it this way, you should probably do it like this:
```suggestion
    pub fn get_range(
        &self,
        addr: &MaybeRelocatable,
        size: usize,
    ) -> Result<Vec<Option<&MaybeRelocatable>>, MemoryError> {
        let addr = if let RelocatableValue(addr) = addr {
            addr
        } else {
            return // some error; we'll write a helper eventually
        }
        let segment = self.data[addr.segment_index];
        let (i, n) = (addr.offset, addr.offset + size);
        if n > segment.len() {
            return // some error
        }
        Ok(Vec::from(segment[i..n]));
    }
```
It's fine to access `data` from within the `memory` module. The problem is only if we do it from the outside.

Reply:
I can't resolve this trivially. Do you think I can change this in a different PR and make a proper iterator?

Reply:
OK.